hapi-fees/src/main/java/com/hedera/services/usage/token/TokenBurnUsage.java

```diff
@@ -36,15 +40,34 @@ public static TokenBurnUsage newEstimate(TransactionBody tokenBurnOp, SigUsage s
 		return new TokenBurnUsage(tokenBurnOp, estimatorFactory.get(sigUsage, tokenBurnOp, ESTIMATOR_UTILS));
 	}
 
+	public TokenBurnUsage givenSubType(SubType subType){
+		this.currentSubType = subType;
+		return this;
+	}
+
 	@Override
 	TokenBurnUsage self() {
 		return this;
 	}
 
 	public FeeData get() {
+		var op = this.op.getTokenBurn();
+
+		if (currentSubType == SubType.TOKEN_NON_FUNGIBLE_UNIQUE) {
+			var bytesToAdd = 0;
+			for (Long o : op.getSerialNumbersList()) {
+				bytesToAdd += o.byteValue();
```

Comment:
‼️ `o.byteValue()` could easily be negative---e.g. `Long.valueOf(37047).byteValue()`...should we just have
```
bytesToAdd = LONG_SIZE * op.getSerialNumbersList().size()
```
or something to that effect?

Reply:
The bug is resolved in an upcoming PR